UnityProject/Assets/Scripts/UI/CameraZoomHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace UI
+{	
+    public class CameraZoomHandler : MonoBehaviour
+    {
+        private int zoomLevel = 1;
+        public Text zoomText;
+
+        void Start()
+        {
+            // Setup initial zoom level by guessing.
+            SetZoomLevel(-1);
+        }
+
+        // Refreshes after setting zoom level.
+        public void Refresh()
+        {
+            // Calculate ratio.
+            double ratio = Screen.height / (double) Screen.width;
+            // Calculate scaling factor. 409600 is a magic number.
+            double scaleFactor = Math.Sqrt(Screen.height * Screen.width / (409600 * ratio));
+            // Calculate orthographic size with full precision and then convert to float precision.
+            Camera.main.orthographicSize = Convert.ToSingle(ratio * 10 * scaleFactor / zoomLevel);
+            // Recenter camera.
+            DisplayManager.Instance.SetCameraFollowPos();
+        }
+
+        public void SetZoomLevel(float zoomLevel)
+        {
+            int integerZoomLevel = (int)zoomLevel;
+		    
+            // Automatically try to determine zoom level if we're feeding it some dumb values, such as when this component initializes.
+            if (integerZoomLevel < 1 || integerZoomLevel > 16)
+            {
+                double ratio = Screen.height / (double) Screen.width;
```

Comment:
You don't need this logic here anymore :) just keep the last 2 lines of this method, Refresh() should have all the logic including zoom level check and text updates. Other than that, it looks fine!